### DISCUSSION TITLE
Start a specified branch in development mode?

### DISCUSSION THREAD

**Discussion by alenart91** [vercel/community#4445] [Help] [answered]
### Summary

I'm trying to setup a live dev site for my project in Vercel. Instead of building I'd like to just run the dev command. I don't see anyway to configure this for the branch.

So for example I have my main branch I use for production that runs the regular build command, and I'd like my dev branch to use the dev command.

Is this possible? If it's not is there a potential workaround? The build command is always going to set NODE_ENV to production, so I'm not sure how that would work.

### Steps to Reproduce

It's not an error.

**Comment by amyegan** [comment, marked as answer]
In general, dev commands are meant to be used _locally_ while developing an app. For example, the [`vercel dev`](https://vercel.com/docs/cli/dev#) command is used locally to replicate the Vercel deployment environment. If you set a [Development Command](https://vercel.com/docs/deployments/configure-a-build#development-command) in your project's settings, that will affect the behavior of `vercel dev`.

The build command is used when deploying an app to the `preview` or `production` [deployment environments](https://vercel.com/docs/deployments/environments). It sets the environment to `production` for deployments from the [Production Branch](https://vercel.com/docs/deployments/git#production-branch) and to `preview` for other branches so the right [environment variables](https://vercel.com/docs/projects/environment-variables#) are used.

**Reply by alenart91** [reply]
Thank you for your response. Since the build command is used for every deployment how would I be able to specify automatically some type of env variable to use for my application logic to use the appropriate configuration? Currently I'm relying on NODE_ENV, but if I can make a custom one such as APP_ENV and set that somewhere I think I can make it work. I'm aware of how to do it manually in the dashboard, but is there a way to set it for each branch in vercel.json or something? I would like to automate it if possible.

**Reply by amyegan** [reply]
There are some [System Environment Variables](https://vercel.com/docs/projects/environment-variables/system-environment-variables#) that are automatically set for each deployment. The `VERCEL_ENV` and `VERCEL_GIT_COMMIT_REF` values may be helpful for your situation.

You can also [create environment variables](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable) in your project's settings. These can be set up for `production`, `preview`, and `development` environments. But you also have the option to [configure env vars for a specific branch](https://vercel.com/changelog/environments-variables-per-git-branch).
